apps/lib/apps.c

```diff
@@ -75,7 +75,7 @@ typedef struct {
     unsigned long mask;
 } NAME_EX_TBL;
 
-static BIO *check_bio(BIO *bp, const char *what, int flags);
+static BIO *check_bio(BIO *bp, const char *what, char mode, int format);
```

Comment:
You are still mixing format and BIO flags - please separate them as well.
This won't be much work ;)